src/NServiceBus.Core/Recoverability/RecoverabilityAction.cs

```diff
@@ -57,6 +60,16 @@ public static Discard Discard(string reason)
             return new Discard(reason);
         }
 
+        /// <summary>
+        /// Creates a custom recoverability action.
+        /// </summary>
+        /// <param name="customAction">The action to perform for the failed message.</param>
+        /// <returns>Custom recoverability action.</returns>
+        public static CustomAction Custom(Func<ErrorContext, IBuilder, IDispatchMessages, Task> customAction)
```

Comment:
Talking to @timbussmann we think that the dispatcher should not be available to users since they should not perform queuing operations since they will run outside of the receive transaction and can therefore cause ghost messages. We already provide ways to move messages to a custom queue via https://docs.particular.net/nservicebus/recoverability/custom-recoverability-policy#implement-a-custom-policy-full-customization

Reply:
Huh? Why is then the transport transaction on the error context? I disagree with that statement.